jsonrpc/filter_manager.go

```diff
@@ -390,6 +390,25 @@ func (f *FilterManager) removeFilterByID(id string) bool {
 	return true
 }
 
+// removeFilterByWs removes the filter with given WS, unsafe against race condition
+func (f *FilterManager) RemoveFilterByWs(ws wsConn) bool {
+	for _, filter := range f.filters {
+		if filter.getFilterBase().ws != ws {
+			continue
+		}
+
+		delete(f.filters, filter.getFilterBase().id)
+
+		if removed := f.timeouts.removeFilter(filter.getFilterBase()); removed {
+			f.emitSignalToUpdateCh()
+		}
+
+		return true
+	}
```

Comment:
This can be optimized - iterating this way is really not performant.

What do you think about adding support for some kind of WS connection ID system? Then, when someone disconnects, you can just delete the connection from the FM with that ID.

Reply:
Fixed dd60f2d672f94444aa9a46fc732c33a46ae61a30 + 865b2c3aed0c45e48c16f21980f4b44c10b319a3

Reply:
I'm not sure if you should use an interface here as the map key, even though the language spec allows it:
>As mentioned earlier, map keys may be of any type that is comparable. The [language spec](https://go.dev/ref/spec#Comparison_operators) defines this precisely, but in short, comparable types are boolean, numeric, string, pointer, channel, and interface types, and structs or arrays that contain only those types. Notably absent from the list are slices, maps, and functions; these types cannot be compared using ==, and may not be used as map keys.

@Kourin1996 ?

Reply:
@Aleksao998 

I like @dbrajovic's suggestion on having this interface give you an ID back:

https://github.com/0xPolygon/polygon-edge/pull/570#pullrequestreview-1039944856

Reply:
Really love when we are playing and optimising PR-s for modules which we can't wait to -rm -rf (af9f7ff6d4b01815b6e7ea3f1aba80c4026dd1bb) :)